src/compiler/scala/tools/nsc/backend/jvm/analysis/BackendUtils.scala

```diff
@@ -324,8 +324,11 @@ abstract class BackendUtils extends PerRunInit {
       bTypesFromClassfile.classBTypeFromParsedClassfile(internalName).info.get.nestedClasses.force
 
     def getClassIfNested(internalName: InternalName): Option[ClassBType] = {
-      val c = bTypesFromClassfile.classBTypeFromParsedClassfile(internalName)
-      if (c.isNestedClass.get) Some(c) else None
+      if (internalName.indexOf('$') < 0) None
```

Comment:
Do $-signs occur primarily close to start or close to end, in the latter case you might want to check `lastIndexOf`.

Reply:
Depends if the inner or outer class has a longer name. The common case is probably a name with no `$`, so we have to look at the whole name most of the time.